### PR TITLE
Switch fetching device info to go over the websocket for RPC devices

### PR DESCRIPTION
**Pull request by bdraco**
Avoid the REST call since we have to create another connection to the device when we do the rest call which will stay open for a bit. By doing everything of the websocket we put less pressure on the device, and setup is a bit faster

### CODE REVIEW

File: aioshelly/rpc_device/device.py [hunk 1]
```diff
@@ -197,23 +197,12 @@ async def _connect_websocket(self) -> None:
...
-                self.aiohttp_session,
-                self.options.ip_address,
-                self.options.device_mac,
-                self.options.port,
-            )
-
-            if self.requires_auth:
-                if self.options.username is None or self.options.password is None:
-                    raise InvalidAuthError("auth missing and required")  # noqa: TRY301
-
```
Comment (bdraco) [inline aioshelly/rpc_device/device.py:210]: This should still raise from the WebSocket instead

**Comment by bdraco** [comment]
I don't think this will work

```
2024-08-27 10:13:47.295 ERROR (MainThread) [homeassistant.components.shelly] Unexpected exception
Traceback (most recent call last):
  File "/Users/bdraco/home-assistant/homeassistant/components/shelly/config_flow.py", line 216, in async_step_credentials
    device_info = await validate_input(
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/homeassistant/components/shelly/config_flow.py", line 109, in validate_input
    await rpc_device.initialize()
  File "/Users/bdraco/home-assistant/venv/lib/python3.12/site-packages/aioshelly/rpc_device/device.py", line 189, in initialize
    await self._connect_websocket()
  File "/Users/bdraco/home-assistant/venv/lib/python3.12/site-packages/aioshelly/rpc_device/device.py", line 202, in _connect_websocket
    self.shelly["auth_domain"],
    ^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/venv/lib/python3.12/site-packages/aioshelly/rpc_device/device.py", line 433, in shelly
    raise NotInitialized
aioshelly.exceptions.NotInitialized
```

File: aioshelly/rpc_device/device.py [hunk 2]
```diff
@@ -285,6 +268,18 @@ async def update_config(self) -> None:
 
     async def _init_calls(self) -> None:
         """Make calls needed to initialize the device."""
+        # _supports_dynamic_components() needs _shelly to be set
+        # so we need to fetch it first. In the future we could
+        # change this to fetch all data in one call once call_rpc_multiple
+        # can be refactored to return errors instead of raising them.
+        self._shelly = await self.call_rpc("Shelly.GetDeviceInfo")
```
Comment (bdraco) [inline aioshelly/rpc_device/device.py:275]: GetDeviceInfo does not require auth per the docs

File: aioshelly/rpc_device/device.py [hunk 3]
```diff
@@ -285,6 +268,18 @@ async def update_config(self) -> None:
 
     async def _init_calls(self) -> None:
         """Make calls needed to initialize the device."""
+        # _supports_dynamic_components() needs _shelly to be set
+        # so we need to fetch it first. In the future we could
+        # change this to fetch all data in one call once call_rpc_multiple
+        # can be refactored to return errors instead of raising them.
```
Comment (bdraco) [inline aioshelly/rpc_device/device.py:274]:
```suggestion
        # Shelly.GetDeviceInfo is the only RPC call that does not
        # require auth, so we must do a separate call here to get
        # the auth_domain/id
```

Adjusted comment as we need to keep it separate

File: aioshelly/rpc_device/device.py [hunk 4]
```diff
@@ -285,6 +268,17 @@ async def update_config(self) -> None:
 
     async def _init_calls(self) -> None:
         """Make calls needed to initialize the device."""
+        # Shelly.GetDeviceInfo is the only RPC call that does not
+        # require auth, so we must do a separate call here to get
+        # the auth_domain/id
```
Comment (bdraco) [inline aioshelly/rpc_device/device.py:273]:
```suggestion
        # the auth_domain/id so we can enable auth for the rest of the calls
```

**Review by thecode — APPROVED** [review]
Nice, tested with Shelly Plus HT and Shelly Plus 1PM Gen3 with and without auth

**Comment by bdraco** [comment]
Did some more testing. All seems good.

**Pull request merged**